moveit_ros/occupancy_map_monitor/src/occupancy_map_monitor.cpp

```diff
@@ -77,106 +71,112 @@ void OccupancyMapMonitor::initialize()
+    RCLCPP_WARN(LOGGER,
+                "Target frame specified but no TF instance specified. No transforms will be applied to received data.");
 
   tree_.reset(new OccMapTree(map_resolution_));
   tree_const_ = tree_;
 
-  XmlRpc::XmlRpcValue sensor_list;
-  if (nh_.getParam("sensors", sensor_list))
-  {
-    try
-    {
-      if (sensor_list.getType() == XmlRpc::XmlRpcValue::TypeArray)
-        for (int32_t i = 0; i < sensor_list.size(); ++i)
-        {
-          if (sensor_list[i].getType() != XmlRpc::XmlRpcValue::TypeStruct)
-          {
-            ROS_ERROR("Params for octomap updater %d not a struct; ignoring.", i);
-            continue;
-          }
-
-          if (!sensor_list[i].hasMember("sensor_plugin"))
-          {
-            ROS_ERROR("No sensor plugin specified for octomap updater %d; ignoring.", i);
-            continue;
-          }
-
-          std::string sensor_plugin = std::string(sensor_list[i]["sensor_plugin"]);
-          if (sensor_plugin.empty() || sensor_plugin[0] == '~')
-          {
-            ROS_INFO("Skipping octomap updater plugin '%s'", sensor_plugin.c_str());
-            continue;
-          }
-
-          if (!updater_plugin_loader_)
-          {
-            try
-            {
-              updater_plugin_loader_.reset(new pluginlib::ClassLoader<OccupancyMapUpdater>(
-                  "moveit_ros_perception", "occupancy_map_monitor::OccupancyMapUpdater"));
-            }
-            catch (pluginlib::PluginlibException& ex)
-            {
-              ROS_FATAL_STREAM("Exception while creating octomap updater plugin loader " << ex.what());
-            }
-          }
-
-          OccupancyMapUpdaterPtr up;
-          try
-          {
-            up = updater_plugin_loader_->createUniqueInstance(sensor_plugin);
-            up->setMonitor(this);
-          }
-          catch (pluginlib::PluginlibException& ex)
-          {
-            ROS_ERROR_STREAM("Exception while loading octomap updater '" << sensor_plugin << "': " << ex.what()
-                                                                         << std::endl);
-          }
-          if (up)
-          {
-            /* pass the params struct directly in to the updater */
-            if (!up->setParams(sensor_list[i]))
-            {
-              ROS_ERROR("Failed to configure updater of type %s", up->getType().c_str());
-              continue;
-            }
-
-            if (!up->initialize())
-            {
-              ROS_ERROR("Unable to initialize map updater of type %s (plugin %s)", up->getType().c_str(),
-                        sensor_plugin.c_str());
-              continue;
-            }
-
-            addUpdater(up);
-          }
-        }
-      else
-        ROS_ERROR("List of sensors must be an array!");
-    }
-    catch (XmlRpc::XmlRpcException& ex)
-    {
-      ROS_ERROR("XmlRpc Exception: %s", ex.getMessage().c_str());
-    }
-  }
+  // TODO rework this in ROS2
+  // XmlRpc::XmlRpcValue sensor_list;
```

Comment:
@JafarAbdi Didn't you have a code snippet where you already replaced XmlRPC ?

Reply:
It's a work-around the way we specify the current parameters, for example for [sensor_3d.yaml](https://github.com/ros-planning/moveit/blob/master/moveit_setup_assistant/templates/moveit_config_pkg_template/config/sensors_3d.yaml) it'll be 

```yaml
/**:
    ros__parameters:
        sensors:  ["sensor1", "sensor2"]
        sensor1:
            sensor_plugin: occupancy_map_monitor/PointCloudOctomapUpdater
            point_cloud_topic: /head_mount_kinect/depth_registered/points
            max_range: 5.0
            point_subsample: 1
            padding_offset: 0.1
            padding_scale: 1.0
            max_update_rate: 1.0
            filtered_cloud_topic: filtered_cloud
        sensor2:
            sensor_plugin: occupancy_map_monitor/DepthImageOctomapUpdater
            image_topic: /head_mount_kinect/depth_registered/image_raw
            queue_size: 5
            near_clipping_plane_distance: 0.3
            far_clipping_plane_distance: 5.0
            shadow_threshold: 0.2
            padding_scale: 4.0
            padding_offset: 0.03
            max_update_rate: 1.0
            filtered_cloud_topic: filtered_cloud
```

This [link](https://answers.ros.org/question/325939/declare-nested-parameter/) inspired me to do so

Reply:
If you think this makes sense, I'll open a PR which load the parameters against @RoboticsYY branch

Reply:
@henningkayser Does this method makes sense .? if you agree with this method I'll open PR fixes it targeting this branch

Reply:
I think it seems a little redundant to have the sensor names extra, but if it makes things easier I agree with this approach.

Reply:
However, I think this belongs to a system wide launch/parameter refactoring, so I would suggest we merge this right away to continue with PSM

Reply:
Yes, I agree with the codebase wide refactoring for XmlRpc